clients/java/signalr/src/main/java/com/microsoft/signalr/HttpHubConnectionBuilder.java

```diff
@@ -22,36 +25,80 @@
         this.url = url;
     }
 
+    /**
+     * Sets the transport to be used by the {@link HubConnection}.
+     *
+     * @param transport The transport to be used.
+     * @return This instance of the HttpHubConnectionBuilder.
+     */
     public HttpHubConnectionBuilder withTransport(Transport transport) {
         this.transport = transport;
         return this;
     }
 
+    /**
+     * Sets the {@link HttpClient} to be used by the {@link HubConnection}.
+     *
+     * @param httpClient The {@link HttpClient} to be used by the {@link HubConnection}.
+     * @return This instance of the HttpHubConnectionBuilder.
+     */
     public HttpHubConnectionBuilder withHttpClient(HttpClient httpClient) {
         this.httpClient = httpClient;
         return this;
     }
 
+    /**
+     * Indicates to the {@link HubConnection} that it should skip the negotiate process.
+     * Note: The Azure SignalR Service require the negotiate step.
```

Comment:
I know we only have WebSockets right now, but maybe we should note this only works with the WebSockets transport so we don't have to remember to update it later.